src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/SourceInfo.cs

```diff
@@ -58,12 +61,17 @@ internal class SourceInfo : ITaintedDataInfo, IEquatable<SourceInfo>
         /// <summary>
```

Comment:
Since all these are getting more complex, should add &lt;remarks&gt; XML documentation, and give examples.

```csharp
/// <remarks>
/// MethodMatcher determines if the outermost tuple applies, based on the method names and arguments.
/// PointsToTaintedTargets.PointsToCheck determines if the inner tuple applies, based on the method invocation's arguments PointsToAbstractvalues.
/// PointsToTaintedTargets.TaintedTargets are the set of tainted targets (arguments / return value).
///
/// Example:
/// (
///   (methodName, argumentOperations) => methodName == "Bar",  // MethodMatcher
///   {
///      (
///         (pointsTos) => true,  // PointsToTaintedTargets.PointsToCheck
///         ImmutableHashSet.Create(new[] { TaintedTargetValue.Return })  // PointsToTaintedTargets.TaintedTargets
///      )
///   }
/// )
///
/// will treat any invocation of the "Bar" method's return value as tainted.
/// </remarks>
```